nextcord/application_command.py

```diff
@@ -1236,217 +864,754 @@ def error(self, callback: ApplicationErrorCallback) -> Callable:
+            name as the value.
+        description_localizations: Dict[Union[:class:`Locale`, :class:`str`], :class:`str`]
+            Description(s) of the command for users of specific locales. The locale code should be the key, with the
+            localized description as the value.
+        callback: Callable
+            Callback to make the application command from, and to run when the application command is called.
+        guild_ids: Iterable[:class:`int`]
+            An iterable list/set/whatever of guild ID's that the application command should register to.
+        dm_permission: :class:`bool`
+            If the command should be usable in DMs or not. Setting to ``False`` will disable the command from being
+            usable in DMs. Only for global commands, but will not error on guild.
+        default_member_permissions: Optional[Union[:class:`Permissions`, :class:`int`]]
+            Permission(s) required to use the command. Inputting ``8`` or ``Permissions(administrator=True)`` for
+            example will only allow Administrators to use the command. If set to 0, nobody will be able to use it by
+            default. Server owners CAN override the permission requirements.
+        parent_cog: Optional[:class:`ClientCog`]
+            ``ClientCog`` to forward to the callback as the ``self`` argument.
+        force_global: :class:`bool`
+            If this command should be registered as a global command, ALONG WITH all guild IDs set.
+        """
+        CallbackWrapperMixin.__init__(self, callback)
+        CallbackMixin.__init__(self, callback=callback, parent_cog=parent_cog)
         self._state: Optional[ConnectionState] = None
+        self.type = cmd_type or ApplicationCommandType(1)
+        self.name: Optional[str] = name
+        self.name_localizations: Optional[Dict[Union[str, Locale], str]] = name_localizations
+        self._description: Optional[str] = description
+        self.description_localizations: Optional[
+            Dict[Union[str, Locale], str]
+        ] = description_localizations
+        self.guild_ids_to_rollout: Set[int] = set(guild_ids) if guild_ids else set()
+        self.dm_permission: Optional[bool] = dm_permission
+        self.default_member_permissions: Optional[
+            Union[Permissions, int]
+        ] = default_member_permissions
+
         self.force_global: bool = force_global
-        self.default_permission: Optional[bool] = default_permission
-        self._guild_ids_to_rollout: Set[int] = set(guild_ids) if guild_ids else set()
-        self._guild_ids: Set[int] = set()
-        # Guild ID is key (None is global), command ID is value.
-        self._command_ids: Dict[Optional[int], int] = {}
 
-    # Simple-ish getter + setters methods.
+        self.command_ids: Dict[Optional[int], int] = {}  # {Guild ID (None for global): command ID}
+        """Command IDs that this application command currently has. Schema: {Guild ID (None for global): command ID}"""
```

Comment:
This docstring isn't displaying correctly.

![image](https://user-images.githubusercontent.com/20955511/169959732-b160c90d-f7b0-491d-891f-3241419e1a30.png)

```suggestion
        self.command_ids: Dict[Optional[int], int] = {}  # {Guild ID (None for global): command ID}
        """Dict[Optional[:class:`int`], :class:`int`]: Command IDs that this application command currently has. Mapping of Guild IDs (None for global) to command IDs."""
```

I think a property method would be cleaner than doing docstrings like this in any case.

Reply:
Fixed ish.